app/assignment/services/assignmentStateService.js

```diff
@@ -873,6 +903,98 @@ class AssignmentStateService {
 		};
 	}
 
+	// Group related TeachingAssignments and generate adjusted priority for looping over in dropdown
+	generateAdjustedPriorities (newState) {
+		if (newState.teachingAssignments) {
```

Comment:
It would be better to exit early if `newState.teachingAssignments` is false, instead of indenting the rest of the function.